src/main/scala/firrtl/backends/verilog/VerilogEmitter.scala

```diff
@@ -849,6 +849,20 @@ class VerilogEmitter extends SeqTransform with Emitter {
             rstring,
             ";"
           )
+        case MemoryFileInlineInit(filename, hexOrBinary) =>
+          if (filename.trim().isEmpty()) {
+            throw EmitterException(s"empty filename not allowed in MemoryFileInlineAnnotation")
+          }
```

Comment:
This very late runtime check (running during Verilog emission) should be promoted to a construction-time requirement on the annotation. That way, you'll get an error as soon as you fail the requirement specified here (like after Chisel elaboration) as opposed to having to wait for the full FIRRTL compile.

(Tangentially, other code here is doing late checks. No reason to follow that pattern here if the error can be produced earlier.)